src/pybind/rados/rados.pyx

```diff
@@ -2799,7 +2799,9 @@ returned %d, but should return zero on success." % (self.name, ret))
             int _flags = flags
 
         with nogil:
-            rados_write_op_operate(_write_op.write_op, self.io, _oid, &_mtime, _flags)
+            ret = rados_write_op_operate(_write_op.write_op, self.io, _oid, &_mtime, _flags)
+        if (ret != 0):
```

Comment:
I'd suggest (b) for this pr. A separate pr removing parens from ifs everywhere in pybind would be great. It's just a historical accident that kept getting copied.